packages/editor/src/components/provider/use-block-editor-settings.js

```diff
@@ -51,11 +69,61 @@ function useBlockEditorSettings( settings, hasTemplate ) {
 				canUser( 'create', 'media' ),
 				true
 			),
+			hasResolvedLocalSiteData: hasFinishedResolvingSiteData,
+			baseUrl: siteData?.url || '',
 		};
 	}, [] );
 
 	const { undo } = useDispatch( editorStore );
 
+	// Temporary home - should this live in `core-data`?
+	const fetchRichUrlData = useCallback(
+		function ( url, fetchOptions = {} ) {
+			if ( ! isURL( url ) ) {
+				return Promise.reject(
+					new TypeError( `${ url } is not a valid URL.` )
+				);
+			}
+
+			// Test for "http" based URL as it is possible for valid
+			// yet unusable URLs such as `tel:123456` to be passed.
+			const protocol = getProtocol( url );
+
+			if (
+				! isValidProtocol( protocol ) ||
+				! protocol.startsWith( 'http' ) ||
+				! /^https?:\/\/[^\/\s]/i.test( url )
```

Comment:
Why do we need this?

Reply:
Cos otherwise things like tel: protocol become valid and we try to fetch them.

Reply:
Oh I mean why we need the last line. Isn't `protocol.startsWith( 'http' )` enough?

Reply:
I'm actually copying if from @talldan's implementation in https://github.com/WordPress/gutenberg/pull/32663

Reply:
Wrong PR? I think it's #11835?

Oh this is interesting. I guess the reason is to ensure there are two `//` after the `http` protocol. So that these will fail:

- `http:example.com`
- `https:/example.com`

However, seems like most major browsers support both above and will interpret them as valid urls.

In the [spec](https://url.spec.whatwg.org/#scheme-relative-special-url-string) though, it's required to have the `//`. Hence, I guess this change is actually preferred. TIL!